### PR TITLE
Adding useEffect for Initial Render on Edit Page 

### PR DESCRIPTION
**Pull request by twentymurial33**
Adding useEffect for Initial Render on Edit Page, "Closes https://github.com/twentymurial33/murials_menu/issues/92"

Displaying the food item that needs to be edited on page so that the user knows what they are editing per `id` provided.

Currently displaying author and title and plain img - 

Screen Shot 2023-04-20 at 9 58 47 AM

### CODE REVIEW

File: client/src/pages/Edit.js [hunk 1]
```diff
@@ -17,17 +18,17 @@ function Edit() {
...
+      const foodResult = await queryAPI({
         url: `http://localhost:5000/menu_items/${id}`,
-        //more functionality with onSuccess Callback
-        onSuccess: (result) => toast.success(result.data, { timeout: 2000 }),
         onError: (result) => toast.error(result.error),
         method: "GET",
       });
+      console.log(data);
+      // console.log(foodResult);
+      dataSet(foodResult);
```
Comment (brentmclark) [inline client/src/pages/Edit.js:28]: I see this approach caused you to change your value props on the inputs below to `data.data.title`, `data.data.author` and `data.data.img`.  If you want, you can *spread* the value of `foodResult` here and return those props to their previous implementation.  

```suggestion
      dataSet(...foodResult);
```

File: client/src/util/fetchFromAPI.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const queryAPI = async (options, toast) => {
     data: null,
     error: null,
   };
-  const response = await fetch(url, method);
+  const response = await fetch(url, { method });
```
Comment (brentmclark) [inline client/src/util/fetchFromAPI.js:17]: Sorry again for not cathing this wayyyyy earlier.  🙁

**Review by brentmclark — APPROVED** [review]
I left a few suggestions for you.  The only thing overly concerning me is the inclusion of `data` in the `useEffect` dependencies list.  With that removed I think you'd be clear to merge this one as is, although I strongly recommend the other proposed changes as well.

File: client/src/pages/Edit.js [hunk 3]
```diff
@@ -17,17 +18,17 @@ function Edit() {
...
-      queryAPI({
+      const foodResult = await queryAPI({
         url: `http://localhost:5000/menu_items/${id}`,
-        //more functionality with onSuccess Callback
-        onSuccess: (result) => toast.success(result.data, { timeout: 2000 }),
         onError: (result) => toast.error(result.error),
         method: "GET",
       });
+      console.log(data);
+      // console.log(foodResult);
```
[the remark this answers is not in the document]
Comment (twentymurial33) [reply client/src/pages/Edit.js:27]: Yes

File: client/src/pages/Edit.js [hunk 4]
```diff
@@ -51,23 +52,24 @@ function Edit() {
...
       console.log(response);
+
       result.error = await response.text();
       toast.error(`Food item not saved.  Error: ${result.error}`);
     }
     return result;
   };
 
-  const navigateToEdit = () => {
+  const HandleNavigate = () => {
```
[the remark this answers is not in the document]
Comment (twentymurial33) [reply client/src/pages/Edit.js:62]: That was a terrible mistake 👎 - not sure why I did that

**Pull request merged**